src/com/interface.rs

```diff
@@ -41,6 +41,24 @@ pub unsafe trait ComInterface: Sized + crate::AbiTransferable {
         }
     }
 
+    fn try_query<Into: ComInterface>(&self) -> Result<Into> {
```

Comment:
I'm confused how `query` above is actually safe now. What if you query for an interface that does not make sense?

Reply:
Yes, if query fails it returns a null pointer. The query function is primarily for the projection to use internally where it knows which queries make sense. So perhaps we need to rename things so that `try_query` has the "good name" and `query` is given a less obvious name.

Reply:
#302